x/valset-pref/keeper_test.go

```diff
@@ -102,6 +91,29 @@ func (suite *KeeperTestSuite) AllocateRewards(ctx sdk.Context, delegator sdk.Acc
 	suite.Require().NotZero(rewardsAfterAllocation[0].Amount)
 }
 
+// PrepareExistingDelegations sets up existing delegation by creating a certain number of validators and delegating tokenAmt to them.
+func (suite *KeeperTestSuite) PrepareExistingDelegations(ctx sdk.Context, valAddrs []string, delegator sdk.AccAddress, tokenAmt sdk.Int) error {
+	for i := 0; i < len(valAddrs); i++ {
+		valAddr, err := sdk.ValAddressFromBech32(valAddrs[i])
+		if err != nil {
+			return fmt.Errorf("validator address not formatted")
+		}
+
+		validator, found := suite.App.StakingKeeper.GetValidator(ctx, valAddr)
+		if !found {
+			return fmt.Errorf("validator not found %s", validator)
+		}
+
+		// Delegate the unbonded tokens
+		_, err = suite.App.StakingKeeper.Delegate(ctx, delegator, tokenAmt, stakingtypes.Unbonded, validator, true)
+		if err != nil {
+			return err
+		}
+	}
```

Comment:
Do we not need to fund delegator with the tokens? Or is that done from the caller?

Reply:
done by the caller. The delegator has to have funds